### ISSUE TITLE
JQL for worklogs fails with large project

### ISSUE THREAD

**Issue by s4nf4n** [shridhar-tl/jira-assistant#337]
### Checklist before you being

- [X] I am sure that I am already using latest version of Jira Assistant
- [X] I had verified that, no similar issues are already reported in [issue tracker](https://github.com/shridhar-tl/jira-assistant/issues)
- [X] I had tried reproducing this issue with with [Web version of Jira Assistant](https://app.jiraassistant.com) to ensure that this issue is not recently fixed and yet to be published
- [X] I had verified that, my query is not answered in [FAQ section of website](https://www.jiraassistant.com/faq)
- [X] I had verified that, my issue is not listed as [known bugs in website](https://www.jiraassistant.com/version-history)

### How do you use Jira Assistant?

Browser extension

### Are you using cloud version of Jira or self hosted (data center / server) of Jira.

Cloud Jira

### Version of Jira Assistant

v2.56

### What browser are you using?

Chrome

### What OS are you using? You need not disclose this if you feel it is irr-relevant for your issue.

Windows

### Bug Description

I am sorry, but I am not entirely sure if this should better be an enhancement. But e.g. Worklog Calendar is not working for me.

The Jira Cloud instance contains a very large project. And the JQL used for fetching worklogs fails with Internal Server Error 500.
JQL used for Worklog Calendar looks like `worklogAuthor in (currentUser()) and worklogDate >= '2023-10-08' and worklogDate < '2023-10-17'`. 

When using the query console in Jira itself it results in the same error. But it is quite easy to compensate for this by adding `AND project!=VERYLARGEPROJECT`.

**Please add functionality to customize worklog JQL.**

### Checklist before you submit

- [X] I have ensured not to paste any confidential information like Jira url, Mail id, etc.
- [X] I have verified my browser console logs and added necessary details (if required)
- [X] I have added required screenshots if the bug is related to UI (as necessary)

**Comment by srtucker** [comment]
I have this issue as well and have been working with Atlassian cloud support for about a month on the same issue. The problem is that the queries times out. They did say that they add some additional indexing to the tables to make it work better but it seems to be hit or miss. I have found that adding `ORDER BY updatedDate DESC` to the JQL makes the query much faster.
It would be great is there was a setting to add additional JQL to the query that the calendar feature uses.

**Comment by shridhar-tl** [comment]
@s4nf4n - Are you still facing this issue or it got resolved by itself?

**Comment by s4nf4n** [comment]
@shridhar-tl Yes still having this issue. Thanks for implementing this. Hope to see this released soon!

**Comment by s4nf4n** [comment]
@shridhar-tl Webversion is working! Thanks alot.
